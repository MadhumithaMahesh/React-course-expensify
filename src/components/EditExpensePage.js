// // // import React from 'react';
// // // import { useParams } from 'react-router-dom';
// // // import {matchPath} from 'react-router-dom'
// // // const match = matchPath(this.props.history.location.pathname,{
// // //   path:'/edit/:param'

// // // })
// // // return(
// // //   <div>
// // //   This is from my edit expense component 
// // // </div>
// // // )
// // import React from 'react';
// // import { connect } from 'react-redux'
// // import ExpenseForm from './ExpenseForm'
// // import {editExpense} from '../actions/expenses'

// // const EditExpensePage = (props) => {
 
// //   return (
// //     <div>
// //       ExpenseForm
// //       <ExpenseForm 
// //       expense = {props.expense}
// //       onSubmit={(expense)=>
// //       {
// //         props.dispatch(editExpense(props.expense.id,expense))
// //         props.history.push('/')
// //       }}></ExpenseForm>
// //     </div>
// //   );
// // };
// // const mapToStateProps = (state,props)=>
// // {
// // return{
// //   expense:state.expenses.find((expense)=>expense.id===props.match.params.id)
// // }
// // }
// // export default connect(mapToStateProps)(EditExpensePage);
// import React from 'react';
// import { connect } from 'react-redux';
// import ExpenseForm from './ExpenseForm';
// import { editExpense, removeExpense } from '../actions/expenses';

// const EditExpensePage = (props) => {
//   return (
//     <div>
//       <ExpenseForm
//         expense={props.expense}
//         onSubmit={(expense) => {
//           props.dispatch(editExpense(props.expense.id, expense));
//           props.history.push('/');
//         }}
//       />
//       <button onClick={() => {
//         props.dispatch(removeExpense({ id: props.expense.id }));
//         props.history.push('/');
//       }}>Remove</button>
//     </div>
//   );
// };

// const mapStateToProps = (state, props) => {
//   return {
//     expense: state.expenses.find((expense) => expense.id === props.match.params.id)
//   };
// };

// export default connect(mapStateToProps)(EditExpensePage);
import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component{
  startEditExpense = (expense)=>
  {
    this.props.startEditExpense(this.props.expense.id,expense)
    this.props.history.push('/');
  }
  startRemoveExpense = (expense)=>
  {
    this.props.startRemoveExpense({id:this.props.expense.id})
    this.props.history.push('/');
  }
  render()
  {
    return(
      <div>
        <h1>Edit expense</h1>
      <ExpenseForm
        expense={this.props.expense}
        onSubmit={this.startEditExpense}
      />
      <button onClick={this.startRemoveExpense}>Remove</button>
    </div>
    )
  }
}
;

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch,props)=>
(
  {
    startEditExpense:(id,expense)=>dispatch(startEditExpense(id,expense)),
    startRemoveExpense:(id)=>dispatch(startRemoveExpense(id))
  }
)

export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);
