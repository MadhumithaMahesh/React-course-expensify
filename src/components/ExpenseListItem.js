// import React from 'react'
// import{Link} from 'react-router-dom'
// import {removeExpense} from '../actions/expenses'
// import { connect } from 'react-redux'

// const ExpenseListItem = ({dispatch,id,description,amount})=>(
//     <div>
//         <Link to ={`/edit/${id}`}>
//         <h1>{description}</h1>
//         </Link>
        
// <h2>{amount}</h2>
// <button onClick={()=>{
//     dispatch(removeExpense({id}))
// }
    
// }>Remove</button>
//     </div>
// )

// export default connect()(ExpenseListItem)

import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>{amount} - {createdAt}</p>
  </div>
);

export default ExpenseListItem;
