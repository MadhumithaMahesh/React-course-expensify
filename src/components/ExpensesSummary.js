import React from 'react'
import {connect} from 'react-redux'
import numeral from 'numeral'
import selectedExpenses from '../selectors/expenses'
import expensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = ({expenseCount,expensesTotal})=>
{
    const expenseWord = expenseCount === 1?'expense':'expenses'
    const formattedExpensesTotal = numeral(expensesTotal/100).format('$0,0.00')
    return(
        <div>
           <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1> 
        </div>
    )
}

const mapToStateProps = (state)=>
{
    const visibleExpenses = selectedExpenses(state.expenses,state.filters)
    return{
        expenseCount:visibleExpenses.length,
        expensesTotal:expensesTotal(visibleExpenses)

    }
}

export default connect(mapToStateProps)(ExpensesSummary)