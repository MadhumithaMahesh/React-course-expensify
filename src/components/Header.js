// import React from 'react';
// import {NavLink} from 'react-router-dom'
// const Header=()=>
    
// (
//   <header>
//   <h1>Expensify</h1>
//   <nav>
//   <NavLink to='/' activeClassName="active" exact={true}>Dashboard</NavLink>
//   <NavLink to='/create' activeClassName="active" >Create</NavLink>
//   <NavLink to='/edit' activeClassName="active" >Edit</NavLink> 
//   <NavLink to='/help' activeClassName="active" >Help</NavLink>
//   </nav>
  
// </header>

// ) 
// export default Header

import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    <NavLink to="/help" activeClassName="is-active">Help</NavLink>
  </header>
);

export default Header;
