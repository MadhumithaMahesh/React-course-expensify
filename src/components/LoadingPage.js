import React from 'react'
 const LoadingPage = ()=>
 (
     <div className="loader">
        <img className="loader-img" src="/images/loader.gif"></img>
     </div>
 )
 export default LoadingPage