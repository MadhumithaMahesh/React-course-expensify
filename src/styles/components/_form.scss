.form{
    display: flex;
    flex-direction: column;
    >*{
        margin-bottom: $s-size;
    }
}

.form-error{
    margin-bottom: $m-size;
    font-style: italic;
}