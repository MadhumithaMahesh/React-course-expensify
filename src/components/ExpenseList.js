// import React from 'react'
// import {connect} from 'react-redux'
// import ExpenseListItem from './ExpenseListItem'
// import SelectedExpenses from '../selectors/expenses'
// import ExpenseListFilter from './ExpenseListFilter'

// const ExpenseList = (props)=>(
//     <div>
// <ExpenseListFilter/>
//         <h1>Expense List</h1>
//     {props.expenses.map((expense)=>
//     {
//         return <ExpenseListItem key= {expense.id} {...expense}/>
        
//     })}
//     </div>
    
// )

// const mapStateToProps =(state)=>
// {
// return{
//     expenses:SelectedExpenses(state.expenses,state.filters)
// }
// }
// export default connect(mapStateToProps)(ExpenseList)

import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.length ===0?(
          <p>No expense</p>
        ):
        (
          props.expenses.map((expense) => {
          return <ExpenseListItem key={expense.id} {...expense} />;
        })
        )
    }
    
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
