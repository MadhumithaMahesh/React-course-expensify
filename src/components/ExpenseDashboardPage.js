// import React from 'react';
// import ExpenseList from './ExpenseList'
// const ExpenseDashboardPage = () => (
//     <div>
//       This is from my dashboard components
//       <ExpenseList></ExpenseList>
//     </div>
//   );

//   export default ExpenseDashboardPage

import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilter';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilter/>
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
