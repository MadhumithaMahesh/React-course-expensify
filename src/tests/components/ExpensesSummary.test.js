import React from 'react'
import {ExpensesSummary} from '../../components/ExpensesSummary'
import {shallow} from 'enzyme'

test('should return summary for one expense',()=>
{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={100}></ExpensesSummary>)
    expect(wrapper).toMatchSnapshot()
})

test('should return summary for many expenses',()=>
{
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={100}></ExpensesSummary>)
    expect(wrapper).toMatchSnapshot()
})


