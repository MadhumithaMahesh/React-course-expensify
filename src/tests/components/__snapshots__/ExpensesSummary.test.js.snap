// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return summary for many expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $1.00
  </h1>
</div>
`;

exports[`should return summary for one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.00
  </h1>
</div>
`;
